### PR TITLE
add new workflows + action

### PR DESCRIPTION
**Pull request by matanshk**
This PR follows the guidelines as described in the [documentation](https://github.com/kubescape/kubescape/tree/master/.github/workflows#readme).

### CODE REVIEW

**Comment by dwertent** [comment]
We should first merge [this](https://github.com/kubescape/workflows/pull/1)

/cc @matanshk

File: .github/workflows/a-pr-scanner.yaml
Comment (dwertent) [inline .github/workflows/a-pr-scanner.yaml:29]: [Question]: Why do we need to install go?

File: .github/workflows/a-pr-scanner.yaml
Comment (dwertent) [inline .github/workflows/a-pr-scanner.yaml:78]: [Request]: Remove the comment

Comment (dwertent) [reply .github/workflows/a-pr-scanner.yaml:78]: [Warninig]: This might break opening PRs from forks

File: .github/workflows/a-pr-scanner.yaml
Comment (dwertent) [inline .github/workflows/a-pr-scanner.yaml:17]: [TODO]: We should run this job from an external job

File: .github/workflows/01-code-review-approved.yaml
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:18]: @vladklokun Notice here the `master` branch

File: .github/workflows/01-code-review-approved.yaml
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:17]: [TODO]: @matanshk add labels to KS repo

File: .github/workflows/b-binary-build-and-e2e-tests.yaml
Comment (dwertent) [inline .github/workflows/b-binary-build-and-e2e-tests.yaml:8]: [TODO]: Remove default if required

File: .github/workflows/b-binary-build-and-e2e-tests.yaml
Comment (dwertent) [inline .github/workflows/b-binary-build-and-e2e-tests.yaml:179]: [Comment]: This cannot change

File: .github/workflows/01-code-review-approved.yaml
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:20]: We should trigger this only if all secrets are found.

File: .github/workflows/02-release.yaml
Comment (dwertent) [inline .github/workflows/02-release.yaml:7]: [TODO]: Remove comment

File: .github/workflows/01-code-review-approved.yaml
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:7]: Add: `paths-ignore:`

File: .github/workflows/01-code-review-approved.yaml
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:39]: [Required] Check that secret is found

**Review by dwertent — CHANGES_REQUESTED** [review]

File: .github/workflows/b-binary-build-and-e2e-tests.yaml
Comment (matanshk) [reply .github/workflows/b-binary-build-and-e2e-tests.yaml:179]: From the workflow perspective, it's can change without breaking nothing (if the systests successfully passed on Ubuntu 22)

File: .github/workflows/01-code-review-approved.yaml
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:51]: [TODO]: We should have a step named `run-merge-to-master`, that step should store the output of the `if` statement so this step will look cleaner

File: .github/workflows/02-release.yaml
Comment (dwertent) [inline .github/workflows/02-release.yaml:31]: [TODO]: We should have a step named `run-binary-build`, that step should store the output of the `if` statement so this step will look cleaner

**Comment by dwertent** [comment]
Please fix the CDO issue :)

File: .github/workflows/01-code-review-approved.yaml
Comment (dwertent) [reply .github/workflows/01-code-review-approved.yaml:7]: @matanshk

**Comment by dwertent** [comment]
@matanshk I'm waiting for the github actions tests to pass before we merge.

@matthyx Notice, here we are overriding the changes you made regarding krew #1022 , we need to make sure to merge the two so we don't lose anything.

**Comment by matthyx** [comment]
> @matanshk I'm waiting for the github actions tests to pass before we merge.
> 
> @matthyx Notice, here we are overriding the changes you made regarding krew #1022 , we need to make sure to merge the two so we don't lose anything.

do you want me to take care of the rebase ?

**Comment by matanshk** [comment]
@dwertent 
All fixed from my side

**Review by dwertent — COMMENTED** [review]
What about the WF we have running now? Are you not deleting the files?

**Review by dwertent — APPROVED** [review]

**Pull request closed without merging**
